apollo-router-core/src/layers/ensure_query_presence.rs

```diff
@@ -33,19 +33,16 @@ where
                         path: Default::default(),
                         extensions: Default::default(),
                     }];
-                    let headers =
-                        std::mem::replace(req.originating_request.headers_mut(), HeaderMap::new())
-                            .into_iter()
-                            .filter_map(|(k, v)| k.map(|k| (k, v)))
-                            .collect();
+
+                    //We do not copy headers from the request to the response as this may lead to leakable of sensitive data
```

Comment:
If we do this, it means we won't be able to implement something in our plugins based on specific headers isn't it ? Like for example with a jwt plugin.

Reply:
It doesn't affect other layers an plugins. It's just for this layer, if it doesn't find a query then it has no reason to bounce back all the incoming headers. In particular a sensitive header may get injected into the originating request.

This got added in #819 possibly by accident. @garypen was there a reason for bouncing back the headers?

Reply:
Oh my bad, I thought it was at the creation of the very first `RouterRequest`. I think this change was coming from me when I had to switch `Headers` representation from a `Vec<(String, String)>` to `Vec<(HeaderName, HeaderValue)>`